### PR TITLE
Node E2E: Avoid printing test result twice.

### PR DESCRIPTION
**Pull request by Random-Liu**
This is a problem since long time ago.

`RunSshCommand` includes the command output to the error. If the command running the test fails, the test output will also be included in the error. [The runner prints both the test output and the error](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/runner/remote/run_remote.go#L270), which leads the test result to be printed twice. (See the [test result](https://storage.googleapis.com/kubernetes-jenkins/logs/kubelet-gce-e2e-ci/10968/build-log.txt) on node tmp-node-e2e-af900a4d-e2e-node-ubuntu-trusty-docker9-v1-image)

This PR changes `RunSshCommand` not to put command output into the error, and leave the caller to decide how to deal with command output when the command fails.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36410)

### CODE REVIEW

**Comment by Random-Liu** [comment]
Mark v1.5 because this fixes the node e2e test framework log for easier debugging.
/cc @saad-ali

**Comment by saad-ali** [comment]
> Mark v1.5 because this fixes the node e2e test framework log for easier debugging.
> /cc @saad-ali

Ack

**Comment by dims** [comment]
LGTM 👍

**Comment by Random-Liu** [comment]
@timstclair Do you wanna take a look? :)

File: test/e2e_node/remote/remote.go [hunk 1]
```diff
@@ -291,7 +288,7 @@ func RunRemote(archive string, host string, cleanup bool, junitFilePrefix string
 
 	glog.Infof("Starting tests on %s", host)
 	output, err = SSH(host, "sh", "-c", cmd)
-
+	// Do not log the output here, let the caller deal with the test output.
 	if err != nil {
```
Comment (timstclair) [inline test/e2e_node/remote/remote.go]: nit: combine this if with the next one

**Review by timstclair — APPROVED** [review]
LGTM

File: test/e2e_node/remote/remote.go (hunk 1 above)
Comment (Random-Liu) [inline test/e2e_node/remote/remote.go]: Will do

**Comment by Random-Liu** [comment]
Addressed comments. Reapply LGTM.

**Pull request merged**